src/lesson/four/MyArrayList.java

```
@@ -0,0 +1,69 @@
+package lesson.four;
+
+public class MyArrayList<T> implements List<T> {
+    transient T[] elementData;
+    private static final int DEFAULT_CAPACITY = 100;
+    private int size = 0;
+
+    public MyArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public MyArrayList(int index) {
+        if (index > DEFAULT_CAPACITY) {
+            elementData = (T[]) new Object[index];
+        } else if (index < DEFAULT_CAPACITY && index >= 0) {
+            elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        } else {
+            throw new IllegalArgumentException("Incorrect argument: " + index);
+        }
+    }
+
+    @Override
+    public void add(T t) {
+        if (size == DEFAULT_CAPACITY) {
+            T[] data = (T[]) new Object[size + 100];
+            System.arraycopy(elementData, 0, data, 0, size);
+            elementData = data;
+        }
+        elementData[size++] = t;
+    }
+
+    @Override
+    public void remove(int index) {
+        if (index < 0 || index > size - 1) {
+            System.out.println("Element does not exist");
```

better throw there custom exception instead of printing